src/IO/Progress.cpp

```diff
@@ -30,7 +47,17 @@ void ProgressValues::write(WriteBuffer & out, UInt64 client_revision) const
 {
     writeVarUInt(read_rows, out);
     writeVarUInt(read_bytes, out);
-    writeVarUInt(total_rows_to_read, out);
+    /// In new TCP protocol we can send total_bytes_to_read without total_rows_to_read.
+    /// If client doesn't support total_bytes_to_read, send approx total_rows_to_read
+    /// to indicate at least approx progress.
+    if (client_revision < DBMS_MIN_PROTOCOL_VERSION_WITH_TOTAL_BYTES_IN_PROGRESS && total_bytes_to_read && !total_rows_to_read)
```

Comment:
It looks strange. Client expect number of rows but we're sending number of bytes, - how this is supposed to work

Reply:
Right now we don't use `total_bytes_to_read ` for progress, because our TCP protocol doesn't support it. But I am going to use `total_bytes_to_read ` instad of `total_rows_to_read ` for better progress in s3/url/file/... table functions (now we calculate approx `total_rows_to_read` using `total_bytes_to_read` inside these table functions). And later we can add new places that will send `total_bytes_to_read` instead of `total_rows_to_read` (becase it can be unknown). And we want old clients to show some progress from new server versions